sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,183 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+from typing import Callable, List
+import uuid
+import asyncio
+import logging
+
+from azure.eventhub import EventPosition, EventHubError
+from azure.eventhub.aio import EventHubClient
+from .checkpoint_manager import CheckpointManager
+from .partition_manager import PartitionManager
+from .partition_processor import PartitionProcessor, CloseReason
+from .utils import get_running_loop
+
+log = logging.getLogger(__name__)
+
+OWNER_LEVEL = 0
+
+
+class EventProcessor(object):
+    def __init__(self, eventhub_client: EventHubClient, consumer_group_name: str,
+                 partition_processor_factory: Callable[[CheckpointManager], PartitionProcessor],
+                 partition_manager: PartitionManager, **kwargs):
+        """An EventProcessor automatically creates and runs consumers for all partitions of the eventhub.
+
+        It provides the user a convenient way to receive events from multiple partitions and save checkpoints.
+        If multiple EventProcessors are running for an event hub, they will automatically balance loading. This feature
+        won't be available until preview 3.
+
+        :param eventhub_client: an instance of azure.eventhub.aio.EventClient object
+        :param consumer_group_name: the consumer group that is used to receive events
+        from the event hub that the eventhub_client is going to receive events from
+        :param partition_processor_factory: a callable (type or function) that is called to return a PartitionProcessor
+        TODO: elaborate an example
```

Comment:
TODO?

Reply:
yes I'm still working on this

Reply:
Updated this part and inserted an example